### PR TITLE
Fixes #3207 Enhance BLIP logging

### LINKED ISSUE #3207: Enhance BLIP logging
Currently BLIP has the option to log messages and frames.  These are currently associated with BLIP+ and BLIP++ logging, but we want the following enhancements:

- [ ] Use a callback w/ a SG log key when this logging is outputted (similar to what's already in place for BLIP level logging).  Requires a blip enhancement.
- [ ] Ensure that setting BLIP++ in SG also enables BLIP+ and BLIP (similar to + handling for other keys)
- [ ] Rename the SG log key as Sync/Sync+/Sync++

### PR DESCRIPTION
**Pull request by tleyden**
Fixes #3207 

- [ ] Merge [go-blip companion PR](https://github.com/couchbase/go-blip/pull/13/files)
- [ ] Update go-blip commit has
- [ ] Merge this PR

### CODE REVIEW

File: rest/blip_sync.go [hunk 1]
```diff
@@ -80,10 +80,10 @@ func (h *handler) handleBLIPSync() error {
 	// Create a BLIP context:
 	blipContext := blip.NewContext()
 	blipContext.Logger = func(fmt string, params ...interface{}) {
-		base.LogTo("BLIP", fmt, params...)
+		base.LogTo("Sync", fmt, params...)
```
Comment (tleyden) [inline rest/blip_sync.go]: This makes it tricky to output `Sync+` prefixed log messages as a result of `blipContext.LogMessages` callbacks and `Sync++` prefixed log messages as a result of  `blipContext.LogFrames` callbacks.

Comment (bbrks) [reply rest/blip_sync.go]: The nicest way I can think of doing this without polluting the go-blip API with SG logging concepts, is changing go-blip to take 3 logger callbacks.

Maybe removing the `LogMessages`/`LogFrames` bools, and use a nil callback func to disable.

E.g: https://play.golang.org/p/z_yfx4A1kFm

Comment (tleyden) [reply rest/blip_sync.go]: @bbrks the plan we've ended up on (so far) was to have blip pass back an extra enum (normal | message | frame), and then the logger callback would use the appropriate log key, depending on the enum val

**Comment by tleyden** [comment]
@adamcfraser ready for review.   There's a drone ci build running.

Also, I updated the initial ticket comments to mention the companion go-blip PR

**Review by adamcfraser — APPROVED** [review]
LGTM, and the go-blip changes look good as well.

**Pull request merged**